src/state/CartProvider.jsx

```diff
@@ -1,103 +1,98 @@
-import React, { useReducer, useContext } from 'react'
+import React, { createContext, useContext, useReducer } from 'react';
 
-// Initialize the context
-const CartContext = React.createContext()
-
-// Definte the default state
 const initialState = {
-  itemsById: {},
-  allItems: [],
-}
-
-// Define reducer actions
-const ADD_ITEM = 'ADD_ITEM'
-const REMOVE_ITEM = 'REMOVE_ITEM'
-const UPDATE_ITEM_QUANTITY = 'UPDATE_ITEM_QUANTITY'
+  cartItems: [],
```

Comment:
We want to keep the `itemsById` and `allItems` because this keeps the state normalized and optimized